addons/website_sale/controllers/main.py

```diff
@@ -485,13 +485,15 @@ def checkout_form_validate(self, mode, all_form_values, data):
         # vat validation
         Partner = request.env['res.partner']
         if data.get("vat") and hasattr(Partner, "check_vat"):
-            check_func = request.website.company_id.vat_check_vies and Partner.vies_vat_check or Partner.simple_vat_check
-            vat_country, vat_number = Partner._split_vat(data.get("vat"))
-            if not check_func(vat_country, vat_number):
-                if country.code and check_func(country.code, data["vat"]):
```

Comment:
Thanks for reply @mart-e 
Sorry, I didn't understand what is the difference of the behaviour between this PR and https://github.com/odoo/odoo/commit/bef9b1114c2e8a15e8a87e1eb5f0451f93ec75d3

I  mean, the original [partner.check_vat](https://github.com/odoo/odoo/blob/2dcf233a2801a6fda0cb03f47aca83182b7803ce/addons/base_vat/models/res_partner.py#L100) supports both format with and without the country code (thanks @qdp-odoo).

Could you help me to understand it, please?